bundles/gcp/src/main/java/org/apache/gravitino/gcs/credential/GCSTokenProvider.java

```diff
@@ -146,6 +146,13 @@ private CredentialAccessBoundary getAccessBoundary(
         CredentialAccessBoundary.newBuilder();
     readBuckets.forEach(
         bucket -> {
+          // Hadoop GCS connector needs to get bucket info
+          AccessBoundaryRule bucketInfoRule =
+              AccessBoundaryRule.newBuilder()
+                  .setAvailableResource(toGCSBucketResource(bucket))
+                  .setAvailablePermissions(Arrays.asList("inRole:roles/storage.legacyBucketReader"))
+                  .build();
+          credentialAccessBoundaryBuilder.addRule(bucketInfoRule);
```

Comment:
This part of the code logic has been verified by https://github.com/apache/gravitino/pull/5974/files#diff-9648afc1108b659efec079440f8e204f9be42501c87bf560cefc2b294a1261c7 and as a fix for permission is not enough for GCS fileset.